### PR TITLE
Code refactoring (#5 Issue)

### LINKED ISSUE #5: Code refactoring
URL in the configuration file
Typing
Dividing classes into files
Code optimization

### PR DESCRIPTION
**Pull request by IgorTest19**
**1)** Speakleash project structure has been modified. Every class was moved from the main `__init__.py` file to the standalone 
    scripts, located within their own modules:

```
speakleash/
├── __init__.py
├── config.json
├── config_loader.py
├── category_manager/
│   ├── __init__.py
│   └── category_manager.py
├── dataset/
│   ├── __init__.py
│   └── dataset.py
├── core/
│   ├── __init__.py
│   └── core.py
└── downloader/
    ├── __init__.py
    └── structure_downloader.py
setup.py
```

**2)** Added a simple config file (`config.json`) containing the URL address. Also, another config file (`config_2.json`) was added to decide which form of presenting and accessing URLs is preferable.
**3)** Added description to every` __init__.py` module file.
**4)** A `config_downloader.py` has been created to make it easier to access the config file with URLs.
**5)** Refactoring the code of the Speakleash class. Docstrings, descriptions, and type hints were added.

TODO:
1) Below modules require further code refactoring, addition of docstrings, descriptions and type hints:
- category_manager/ (done)
- dataset/ (done)
- downloader/ (done)
2) Adding proper exceptions (with logging)
3) Adding fixes to known issues
4) Adding tests

### CODE REVIEW

File: setup.py
Comment (pawelkiszczak) [inline setup.py]: Shouldn't the version number be updated?

File: speakleash/category_manager/category_manager.py
Comment (pawelkiszczak) [inline speakleash/category_manager/category_manager.py:86]: Can we go with `except Exception as e` to know the exact error code?

File: speakleash/config_2.json
Comment (pawelkiszczak) [inline speakleash/config_2.json:1]: File naming to be more specific or corresponding to a given part of the package.

**Review by pawelkiszczak — COMMENTED** [review]
Everything looks excellent and no significant bugs were discovered.

File: speakleash/category_manager/category_manager.py
Comment (IgorTest19) [reply speakleash/category_manager/category_manager.py:86]: Exceptions and errors were left unchanged here. They are being modified but code is gonna be delivered in the next, bigger pull request. There are also other things under development but we needed modified main code at first.

File: speakleash/config_2.json
Comment (IgorTest19) [reply speakleash/config_2.json:1]: Only one of these two config files will remain (config.json / config_2.json). It is a good suggestion to rename it like 'urls_config' or something like that, because over time, there will be more config files and there should be also created a directory for them.

File: setup.py
Comment (IgorTest19) [reply setup.py]: Version number will be updated in the day of the deployment.

File: speakleash/category_manager/category_manager.py
Comment (pawelkiszczak) [reply speakleash/category_manager/category_manager.py:86]: Okay, approved then.

File: speakleash/config_2.json
Comment (pawelkiszczak) [reply speakleash/config_2.json:1]: Therefore please rename the config files as what they are used for for future iterations. For now I can go with approval.

**Review by pawelkiszczak — APPROVED** [review]
Approved after a detailed description from the author

**Comment by IgorTest19** [comment]
Before merging to the `main` branch, the new version of the package should be set in the `setup.py` file (the current version is `0.2.3`). Due to the fact, that this pull request is a part of 3 large pull requests of a complex code refactoring process and still maintains backwards compatibility, it is suggested to set the new minor number value, from `0.2.3` to `0.3.3`. The next pull request should be set as `0.4.3`. The last PR will the final one of this code refactor process, so the package version will be set to `1.0.0`.

**Pull request merged**
